### DISCUSSION TITLE
Metadata and callback function

### DISCUSSION THREAD

**Discussion by kodjo80** [pschatzmann/ESP32-A2DP#124] [Q&A] [answered]
Hello, I tried your library and it's very useful and easy to use. However I don't understand how to implement metadata callback function.
As I understand it I have to declare avrc_metadata_callback function before void setup() and i have to insert 
`a2dp_sink.set_avrc_metadata_callback(avrc_metadata_callback);` before starting a2dp sink.
How do I get the metadata afterwards. I expect to see info on the artist and other metadata on serial monitor when there is a change, but in my serial monitor there is no such data. Am I missing something or I don't understand callback function at all?

**Comment by pschatzmann** [comment, marked as answer]
Your expectation is correct. If the bluetooth source (e.g. mobile app) is sending metadata, the callback is executed and the metadata is provided in the parameters of the call. In the example this info is just printed in the log. 

So if you don't get anything, I assume your application is not sending anything. 
I suggest that you double check with Youtube

**Reply by kodjo80** [reply]
That's it. Thank you.
I found that the artist name is not displayed properly if there is a comma in artist name, but I will dig deeper if I can sort it out.
